File: python-algorithm/leetcode/problem_1791.py
"""1791. Find Center of Star Graph
https://leetcode.com/problems/find-center-of-star-graph/
"""
from typing import List


class Solution:
    def findCenter(self, edges: List[List[int]]) -> int:
        candidates = edges[0]
        for u, v in edges:
            if u != candidates[0] and v != candidates[1]:
                return candidates[1]
            if u != candidates[0] and u != candidates[1]:
                return candidates[0]
        return 0

File: java-algo/src/main/java/com/leetcode/Problem1791.java
package com.leetcode;

import java.util.HashMap;
import java.util.Map;

/**
 * 1791. Find Center of Star Graph
 * https://leetcode.com/problems/find-center-of-star-graph/
 */
public class Problem1791 {
    public int findCenter(int[][] edges) {
        Map<Integer, Integer> counter = new HashMap<>();
        for (int[] edge : edges) {
            int u = edge[0], v = edge[1];
            if (counter.containsKey(u)) {
                return u;
            }
            counter.put(u, 1);
            if (counter.containsKey(v)) {
                return v;
            }
            counter.put(v, 1);
        }
        return 0;
    }
}
